tsconfig.json

```diff
@@ -0,0 +1,33 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "inlineSources": true,
+    "isolatedModules": true,
+    "jsx": "react",
+    "lib": ["dom", "es2020"],
```

Comment:
These settings were copied from the [`tsconfig.json`](https://github.com/MetaMask/controllers/blob/81bcbae1c4abe71a9b07819606e155a33ec4a6b4/tsconfig.json) in `controllers` but then adjusted to account for JSX as well as browser globals. ES2020 was chosen because Node v14 supports it, so that should make it less surprising for developers who wish to be able to write frontend code in a similar fashion as Node code. (The JavaScript that TypeScript generates will end up getting processed by Babel in the build system so we should be safe here.)